exercises/concept/captains-log/.meta/design.md

```diff
@@ -19,3 +19,21 @@
 - `strings`: know how to use string formatting.
 - `numbers`: know how to apply basic mathematical operators.
 - `arrays`: know how to retrieve array elements by their index.
+
+## Analyzer
+
+This exercise could benefit from the following rules in the [analyzer]:
+
+- `actionable`: If the solution hardcodes the number of planet classes in `randomPlanetClass`, instruct the student to use `length` method instead.
+- `informative`: If the solution uses [`RandomGenerator.nextDouble(double)`][nextDouble(double)], inform the student that they could use [`RandomGenerator.nextDouble(double, double)`][nextDouble(double, double)] to achieve a clearer solution.
+- `informative`: If the solution uses [`RandomGenerator.nextInt(int)`][nextInt(int)], inform the student that they could use [`RandomGenerator.nextInt(int, int)`][nextInt(int, int)] to achieve a clearer solution.
```

Comment:
Note that this depends on the Java version used by the student. `nextInt(int)` is available in Java 8 and above, but `nextInt(int,int)` is not available until Java 17.

So this feedback may confuse students who are using older Java versions because their code will not compile after applying the suggestion.

Reply:
This is because of the exercism cli, right? because on the ui you cannot choose the java version, I never used it before, and I guess we cannot check for the java version priori to leave the comment, so well maybe leave only the actionable for the length and the string,format ones

Reply:
Yeah I don't have a clear-cut answer for this either. I suppose it's safe to assume that when a solution uses `RandomGenerator.nextDouble(double, double)` they must be using Java 17 or higher, but other than that I don't think we can manage it right now.

Keep in mind though that the design document contains the _intentions_ behind the analyzer, that doesn't necessarily mean that everything in it should be (or will be) implemented. IMO it's fine if this `design.md` describes that the analyzer could point solutions to the `RandomGenerator` methods, but only when using Java 17 and above. If we ever come to a point where we can determine this in the analyzer we have a record here reminding us to add those rules.

Reply:
Okey so I think with this clarification that I added to those comments we are great then.